app/src/main/java/com/eveningoutpost/dexdrip/Models/blueReader.java

```diff
@@ -47,20 +69,27 @@ public static boolean isblueReaderPacket(byte[] buffer) {
         }
         if (bufferstring.startsWith("not ready for") ) { //delete the trans_failed, because its normal only if the bluereader could not read the sensor.
             counterHibernated++;
-            Log.e(TAG, "Found blueReader in a ugly State (" + counterHibernated + "/5), send hibernate to reset! If this does not help in the next 5 Minutes, then turn the bluereader manually off and on!");
-            if (counterHibernated > 4) {
-                Log.wtf(TAG, "Ugly state not resolveable. Please restart the BlueReader! Sending shutdown to blueReader!");
+            Log.e(TAG, "Found blueReader in a ugly State (" + counterHibernated + "/3), send hibernate to reset! If this does not help in the next 5 Minutes, then turn the bluereader manually off and on!");
+            if (counterHibernated > (int)2) {
+                Log.wtf(TAG, "Ugly state not resolveable. Bluereader will be shut down! Please restart it!");
                 Home.toaststatic("BlueReader ugly state not resolveable, bluereader will be shut down. Please restart it!");
-                return new byte[]{0x6B};
+                if (!Pref.getBooleanDefaultFalse("blueReader_suppressuglystatemsg")) {
+                    Notifications.RiseDropAlert(xdrip.getAppContext(),true,"BlueReader Alarm", xdrip.getAppContext().getString(R.string.bluereaderuglystate),1);
+                }
+                return shutdown;
+            } else {
+                Home.toaststatic("Found blueReader in a ugly State, send hibernate to reset!");
```

Comment:
I cannot see that the tests has been updated. Please do a pull from upstream (pull upstream master in your branch) to get the latest version and ensure the tests are OK before we submit. I can assist on fixing the tests if you need help.

Reply:
I've updated the tests and they are ok now.

Reply:
looks good.